src/CVQualifiers.cpp

```diff
@@ -284,6 +284,26 @@ CVQualifiers::random_qualifiers(const Type* t, Effect::Access access,
 	return random_qualifiers(t, access, cg_context, no_volatile, RegularConstProb, RegularVolatileProb);
 }
 
+static bool is_volatile_ok_on_one_level(const Type* t)
+{
+	if (!CGOptions::lang_cpp()) return true;
+	bool volatile_ok = (t->eType != eStruct && t->eType != eUnion) || (t->has_assign_ops());
+	// Union with a struct field: we can't make it volatile, or we won't be able to assign to/from that field
+	if (volatile_ok && (t->eType == eUnion)){
+		for (const auto& field : t->fields){
```

Comment:
Can we avoid using range-based for loop at the moment? We haven't enabled C++11 build for csmith.